rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -776,8 +776,7 @@ private void abortLRA() {
             // just update the status for now and wait for the external LRA close and cancel from there instead
             status = LRAStatus.Cancelling;
```

Comment:
Are you certain this assumption is safe as it is just in memory state? What if another process happened to load the same LRA from disk? Would it be worth writing out this state change or can it be reliably calculated during rehydration?

Reply:
It can be reliably calculated at re-hydrate time. The relevant checks are:
https://github.com/jbosstm/narayana/blob/master/rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java#L86

and

https://github.com/jbosstm/narayana/blob/master/rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java#L252

I raised https://issues.jboss.org/browse/JBTM-3212 to make sure it is implemented correctly.

Reply:
Thanks

Reply:
Would you agree that https://issues.jboss.org/browse/JBTM-3212 is a blocker? Otherwise this process could be deciding to cancel the LRA and a second process that starts and loads from the object store might decide the complete the transaction?

Reply:
Well it is no different from the current code but I agree it is a blocker - I will fix it.

Reply:
But I think it is better if I fix it in a separate PR. Is that okay @tomjenkinson

Reply:
Thanks Mike